utils/ssh.py

```diff
@@ -241,11 +274,11 @@ def run_command(self, command, timeout=RUNCMD_TIMEOUT, reraise=False, ensure_hos
         except socket.timeout as e:
             logger.error("Command `%s` timed out.", command)
             logger.exception(e)
-            logger.error("Output of the command before it failed was:\n%s", output)
+            logger.error("Output of the command before it failed was:\n%s", ''.join(output))
             raise
 
         # Returning two things so tuple unpacking the return works even if the ssh client fails
-        return SSHResult(1, None)
+        return SSHResult(1, '')
```

Comment:
It should have always returned string anyway. Maybe I would change the $? to -1 to detect an error.